Dockerfile

```diff
@@ -1,6 +1,12 @@
 FROM ruby:2.5.1
-RUN apt-get update && apt-get install -y build-essential libpq-dev nodejs
+RUN apt-get update && apt-get install -y build-essential libpq-dev
+RUN curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN apt-get update && apt-get install -y nodejs
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+RUN apt-get update && apt-get install -y yarn
```

Comment:
All above commands can be added using a single RUN instruction so that a single build layer is built which would help in reducing image size and caching benefits.

For example,

```Dockerfile
RUN apt-get update && apt-get install -y build-essential libpq-dev \
    && curl -sL https://deb.nodesource.com/setup_10.x | bash - \
    && apt-get update && apt-get install -y nodejs \
    && ...
```

What do you think?

Reply:
@vishaltelangre  I tried it, but with these set of commands it did not work. It is easy to club commands like just `apt-get` in single command but not these commands. But we can surely do it after more trial and testing. Considering this PR mainly intended to support blog story, I did not spend much time. But yes, if we want to take this up on priority, we can work on lines of production image building techniques